src/assignment/components/AssignmentAdminFormEditable.tsx

```diff
@@ -0,0 +1,203 @@
+import {
+	Column,
+	Container,
+	Form,
+	FormGroup,
+	Grid,
+	Spacer,
+	TagInfo,
+	TagsInput,
+	TextArea,
+	TextInput,
+} from '@viaa/avo2-components';
+import { Avo, PermissionName } from '@viaa/avo2-types';
+import { isNil } from 'lodash-es';
+import React, { Dispatch, FC, SetStateAction } from 'react';
+import { UseFormSetValue } from 'react-hook-form';
+
+import { ContentPicker } from '../../admin/shared/components/ContentPicker/ContentPicker';
+import { PickerItem } from '../../admin/shared/types';
+import { PermissionService } from '../../authentication/helpers/permission-service';
+import { QualityLabel } from '../../collection/collection.types';
+import { formatTimestamp, getFullName } from '../../shared/helpers';
+import withUser, { UserProps } from '../../shared/hocs/withUser';
+import { useGetQualityLabels } from '../../shared/hooks';
+import useTranslation from '../../shared/hooks/useTranslation';
+
+interface AssignmentAdminFormEditableProps {
+	assignment: Avo.Assignment.Assignment;
+	setAssignment: Dispatch<SetStateAction<Avo.Assignment.Assignment>>;
+	setValue: UseFormSetValue<Avo.Assignment.Assignment>;
+}
+
+const AssignmentAdminFormEditable: FC<AssignmentAdminFormEditableProps & UserProps> = ({
+	assignment,
+	user,
+	setAssignment,
+	setValue,
+}) => {
+	const { tText } = useTranslation();
+	const { data: qualityLabels, isLoading } = useGetQualityLabels();
+
+	const transformQualityLabelsToTagInfo = (labels: QualityLabel[]): TagInfo[] => {
+		return labels.map((label: QualityLabel) => ({
+			label: label.description,
+			value: label.value,
+		}));
+	};
+
+	const getAssignmentLabels = (): TagInfo[] => {
+		if (!qualityLabels) {
+			return [];
+		}
+		const labelIds = ((assignment.quality_labels || []) as Avo.Assignment.QualityLabel[]).map(
+			(item: any) => item.label
+		);
+		const filteredQualityLabels = qualityLabels.filter((qualityLabel) =>
+			labelIds.includes(qualityLabel.value)
+		);
+
+		return transformQualityLabelsToTagInfo(filteredQualityLabels);
+	};
+
+	const handleQualityLabelChange = (labels: TagInfo[]) => {
+		const newQualityLabels = labels.map(
+			(label) =>
+				({
+					assignment_id: assignment.id,
+					label: label.value,
+				} as Avo.Assignment.QualityLabel)
+		);
+
+		(setValue as any)('quality_labels', newQualityLabels, {
+			shouldDirty: true,
+			shouldTouch: true,
+		});
+		setAssignment((prev) => ({
+			...prev,
+			quality_labels: newQualityLabels,
+			blocks: (prev as Avo.Assignment.Assignment)?.blocks || [],
+		}));
+	};
+
+	const owner: PickerItem | undefined = assignment.profile
+		? {
+				label: `${assignment.profile.user.first_name} ${assignment.profile.user.last_name} (${assignment.profile.user.mail})`,
+				type: 'PROFILE',
+				value: assignment.profile.id,
+		  }
+		: undefined;
```

Comment:
put computed values at the top after the regular variables and before the getters